pkg/skbn/abs.go

```diff
@@ -0,0 +1,184 @@
+package skbn
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/azure-pipeline-go/pipeline"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+)
+
+var err error
+
+func validateAbsPath(pathSplit []string) error {
+	if len(pathSplit) >= 1 {
+		return nil
+	}
+	return fmt.Errorf("illegal path: %s", filepath.Join(pathSplit...))
+}
+
+func initAbsVariables(split []string) (string, string, string) {
+	account := split[0]
+	container := split[1]
+	path := filepath.Join(split[2:]...)
+
+	return account, container, path
+}
+
+func getNewPipeline() (pipeline.Pipeline, error) {
+	accountName, accountKey := os.Getenv("AZURE_STORAGE_ACCOUNT"), os.Getenv("AZURE_STORAGE_ACCESS_KEY")
+
+	if len(accountName) == 0 || len(accountKey) == 0 {
+		log.Fatal("Either the AZURE_STORAGE_ACCOUNT or AZURE_STORAGE_ACCESS_KEY environment variable is not set")
+	}
+
+	credential, err := azblob.NewSharedKeyCredential(accountName, accountKey)
+
+	if err != nil {
+		log.Fatal("Invalid credentials with error: " + err.Error())
+	}
+	pl := azblob.NewPipeline(credential, azblob.PipelineOptions{})
+
+	return pl, err
+}
+
+func getServiceURL(pl pipeline.Pipeline, accountName string) (azblob.ServiceURL, error) {
+	URL, err := url.Parse(
+		fmt.Sprintf("https://%s.blob.core.windows.net/", accountName))
+
+	surl := azblob.NewServiceURL(*URL, pl)
+	return surl, err
+}
+
+func getContainerURL(pl pipeline.Pipeline, accountName string, containerName string) (azblob.ContainerURL, error) {
+	URL, err := url.Parse(
+		fmt.Sprintf("https://%s.blob.core.windows.net/%s", accountName, containerName))
+
+	curl := azblob.NewContainerURL(*URL, pl)
+	return curl, err
+}
+
+func getBlobURL(curl azblob.ContainerURL, blob string) (azblob.BlockBlobURL, error) {
+	return curl.NewBlockBlobURL(blob), err
+}
+
+func createContainer(ctx context.Context, pl pipeline.Pipeline, curl azblob.ContainerURL) (*azblob.ContainerCreateResponse, error) {
+	cr, err := curl.Create(ctx, azblob.Metadata{}, azblob.PublicAccessNone)
+	return cr, err
+}
+
+func listContainers(ctx context.Context, surl azblob.ServiceURL) ([]azblob.ContainerItem, error) {
+	lc, err := surl.ListContainersSegment(ctx, azblob.Marker{}, azblob.ListContainersSegmentOptions{})
```

Comment:
If a user has authorization to upload/download from a container, does he/she also have the privileges to list the existing containers? Or are these separate? For example, in AWS, you can have access to a bucket (container), but not necessarily have access to list all the buckets.

If these are indeed separate permissions, we shouldn`t assume that this user has them both.

Reply:
I use storage account level permissions at this moment. We can try to per-container sas token, but it will require container existance and sas token generation will require subscription based perms.

Reply:
I will investigate this opportunity

Reply:
Left it for the future improvements